### DISCUSSION TITLE
Use PSR-3 for the Logger

### DISCUSSION THREAD

**Discussion by litzinger** [ExpressionEngine/ExpressionEngine#2996] [Feature Requests]
Currently EE's logger defaults to logging to the database, which is fine in some cases, but more robust applications would benefit from additional logging options, for example, the file system. In our case, we use SumoLogic to scrape log files from multiple web servers. SumoLogic is a more robust logging option that a basic database table (which increases the size of the production db).

I propose that the ee()->logger class be refactored to implement the PSR-3 interface, and also provide a way for developers to extend or define a different driver. For example, I could create my own logging class using the PSR-3 standard that writes to the filesystem, perhaps using Monolog, then in the config.php file tell EE to use my class as in the logging service.

Or, just switch EE's entire logging service to Monolog :) Either way, providing a way for 3rd party developers to change where EE sends its logs would be nice.

https://www.php-fig.org/psr/psr-3/
https://github.com/Seldaek/monolog

**Comment by eric-lamb62** [comment]
Yes please and thank you. I can really see the benefits of this.

**Comment by litzinger** [comment]
I should uncheck the "I am capable" box... I don't think I have enough time to tackle this on the side anymore :/

**Comment by eric-lamb62** [comment]
Oh sure thing @litzinger I can take over the checkbox there :)
